libraries/AP_NavEKF3/AP_NavEKF3_core.h

```diff
@@ -628,6 +628,19 @@ class NavEKF3_core : public NavEKF_core_common
         Vector3f accel_bias;
     } inactiveBias[INS_MAX_INSTANCES];
 
+    // Specify source of data to be used for a partial state reset
+    // Checking the availability and quality of the data source specified is the responsibility of the caller
+    enum resetDataSource {
```

Comment:
I was thinking this would be nicer as an `enum class` earlier today.  Not relevant for this PR.

I was *also* thinking that there should be two *different* enum classes - we shouldn't share the same set of valid data sources between position and velocity.  Again, not really relevant for this PR.

Reply:
Thank you @peterbarker . Should I change it in this PR or create another PR for this?

Reply:
I would just keep it as one enum for now.  I think it's not totally clear if multiple enums would reduce programming errors or not.